test/aem/aemupload.test.js

```diff
@@ -284,4 +284,209 @@ describe('AEM Upload', function() {
+        folderPath: assetPath,
+        files: [{
+            fileName: assetName,
+            mimeType: 'image/jpeg',
+            uploadToken: 'upload-token',
+            uploadURIs: [
+                `${HOST}/part`
+            ],
+            minPartSize: 10,
+            maxPartSize: 100
+        }]
+    };
+    const initRaw = JSON.stringify(initResponse);
+    
+    nock(HOST)
+        .post(`${encodePathParts(assetPath)}.initiateUpload.json`, `fileName=${getFormEncoding(assetName)}&fileSize=15`)
+        .reply(201, initRaw, {
+            'Content-Length': initRaw.length
+        });
+
+    nock(HOST, {
+        reqheaders: {
+            'content-length': 15,
+            'content-type': 'image/jpeg',
+            partHeader: 'test'
+        }
+    })
+        .put('/part', 'hello world 123')
+        .reply(201);
+
+    nock(HOST)
+        .post(`${encodePathParts(assetPath)}.completeUpload.json`, (body) => {
+            const {
+                fileName,
+                fileSize,
+                mimeType,
+                createVersion,
+                versionLabel,
+                versionComment,
+                replace,
+                uploadToken,
+                uploadDuration
+            } = body;
+            return fileName === decodeURIComponent(assetName) && fileSize === "15" && mimeType === "image/jpeg" &&
+                createVersion === "true" && versionLabel === "versionLabel" && versionComment === "versionComment" &&
+                replace === "false" && uploadToken === "upload-token" && uploadDuration;
+        })
+        .reply(200, '{}');
+
+    const aemUpload = new AEMUpload();
+    const events = {
+        filestart: [],
+        fileprogress:[],
+        fileend: []
+    };
+    aemUpload.on('filestart', (data) => {
+        events.filestart.push(data);
+    });
+    aemUpload.on('fileprogress', (data) => {
+        events.fileprogress.push(data);
+    });
+    aemUpload.on('fileend', (data) => {
+        events.fileend.push(data);
+    });
+    await aemUpload.uploadFiles({
+        uploadFiles: [{
+            fileUrl: `http://test-aem-upload-201${assetPath}/${assetName}`,
+            filePath: testFile,
+            fileSize: 15,
+            createVersion: true,
+            versionLabel: 'versionLabel',
+            versionComment: 'versionComment',
+            multipartHeaders: { partHeader: 'test' },
+        }],
+        headers: {},
+        concurrent: true,
+        maxConcurrent: 5,
+        preferredPartSize: 7,
+        supportSpecialCharacters,
+    });
+
+    try {
+        await fs.unlink(testFile);
+    } catch (e) { // ignore cleanup failures
+        console.log(e);
+    }
+
+    const fileEventData = {
+        fileName: decodeURIComponent(assetName),
+        fileSize: 15,
+        targetFolder: decodeURIComponent(assetPath),
+        targetFile: decodeURIComponent(`${assetPath}/${assetName}`),
+        sourceFolder: __dirname,
+        sourceFile: testFile
+    };
```

Comment:
Why is it decoding the `assetName`, `assetPath`?

Reply:
This decoding is also done to support the [unit test where the target path is already URL encoded](https://github.com/adobe/node-httptransfer/pull/130/files#diff-42bb1c8fc4d7014d2ca5b6b5e77959d431bc7e600f27295d81d60b9f58a61476R306). The decoding here in the test code matches decoding performed on [target paths and filenames](https://github.com/adobe/node-httptransfer/blob/17007fe84fc6d933b0849b9502a3b09cd69d3131/lib/util.js#L203) when `TransferAsset` [eventData is requested](https://github.com/adobe/node-httptransfer/blob/17007fe84fc6d933b0849b9502a3b09cd69d3131/lib/asset/transferasset.js#L201).